modules/javafx.controls/src/main/java/javafx/scene/control/Spinner.java

```diff
@@ -785,6 +785,22 @@ static int wrapValue(int value, int min, int max) {
         return r;
     }
 
+    /*
+     * Convenience method to support wrapping values around their min / max
+     * constraints. Used by the SpinnerValueFactory implementations when
+     * the Spinner wrapAround property is true.
+     */
+    static int wrapValue(int value, int min, int max, boolean wrapUp) {
```

Comment:
This is essentially a throw-away function. While it does handle the specific case in the bug, namely that of incrementing or decrementing a positive number of steps of an `amountToStepBy` that is also positive as long as `currentVal + nsteps * amountToStepBy <= `2 * max`.

It does it by taking a boolean parameter and doing a simple subtractions, which is not how it eventually needs to be fixed. It won't handle the case of wrapping around by more than `max-min+1` nor will it handle the case of wrapping around with a negative `amountToStepBy`.

Also, I did find one case where it will yield differently incorrect behavior from today. in the case where min = 0, and you step by an amount that puts the new value at `newValue > 2 * max` the old code would at least compute an almost-correct value using `newVal % max`. The new code will cause it to be clamped at max.